### ISSUE TITLE
Launcher: An error occurred while updating the launcher

### ISSUE THREAD

**Issue by Simba-Mupfunya** [xenko3d/xenko#1]
The launcher crashes with the message “An error occurred while updating the launcher” even after a complete uninstall.

**Comment by tebjan** [comment]
works fine for me, maybe something on your end?

**Comment by tebjan** [comment]
i have updated from a previous version... did you have a version installed before?

**Comment by Simba-Mupfunya** [comment]
Yes, now they are all gone naturally, but it seems as if it wants to update something that's not there even after an uninstall

**Comment by Simba-Mupfunya** [comment]
@Kryptos-FR Belgium

**Comment by Kryptos-FR** [comment]
Hmm. No, I don't think we ever had an issue with Europe.

**Comment by Simba-Mupfunya** [comment]
i know, im stumped, i dont quite understand why its giving an upgrade error message now that everything is also uninstalled.. Maybe i need a new installer that is not the one publicly available.

**Comment by TomGroner** [comment]
I am receiving the same error as @Simba-Mupfunya. I am from the US.

**Comment by xen2** [comment]
@TomGroner @Simba-Mupfunya I will take a look if I see anything wrong with that code (might not be easy without more info)
If you happen to be able to attach a Visual Studio debugger to the process and grab some caught exceptions & stacktrace, that would be helpful!

**Comment by TomGroner** [comment]
I also tried uninstalling, and re-installing to a new location. PS sorry if that picture isn't helpful, I realize it's probably pointing at the first place you would be looking :(

**Comment by TomGroner** [comment]
@xen2 is the source code for the launcher not in the repo? I was going to try compiling it from source to see if it gave the same error. I couldn't seem to find it and then I noticed the .gitignore was updated to ignore the /sources/launcher/ directory.

**Comment by TomGroner** [comment]
I was able to resolve this problem after taking these steps:
1) Uninstalled Xenko
2) Re-start computer
3) Reinstall Xenko
4) Removing the following folder: C:\Users\[Username]\AppData\Local\Silicon Studios\

If I had to place a bet step 4 is probably the only necessary step.

**Comment by TomGroner** [comment]
As an added piece of information, I have my own personal NuGet server that I keep some of my test packages on when tinkering. I attempted to install the Xenko 2.1 beta version and it failed with a message that the VM that I run my NuGet server from was not found. When I turned it on I was able to install the older Xenko version without issue. 

I'm wondering if that error, along with the fact that Xenko 3.0 creates and registers it's own local Nuget, could have been the issue more than deleting the old Silicon Studios directory in AppData? I would test the theory with my VM running, but I permanently deleted the old AppData folder so I don't have the pieces to re-create the conditions.

**Comment by xen2** [comment]
@TomGroner thanks for the details, it definitely could be something related to Nuget since we use it much more extensively now. Will check if something stands out.

So far launcher is still closed source (I was considering an asset store but that's much later down the road), but maybe I will reconsider opensourcing (maybe not MIT?) it just to make everybody's life easier and contributions possible.

**Comment by TomGroner** [comment]
@xen2 Yeah I think the issue might a need to be specific about which NuGet server being is critical enough to halt on. The primary nuget.org server, the Xenko internal ones, those are critical, But SomeRandomOrg.org could be ignored. I have NuGet sources mapped such as my personal one and one for work that isn't available unless I have a VPN session active. I am curious if @Simba-Mupfunya has something similar setup in his environment.

**Comment by RUSshy** [comment]
Work fine for me too
I first uninstalled the previous version from the launcher, then i installed 3.0

**Comment by Simba-Mupfunya** [comment]
OMG @TomGroner, you nailed it. My setup does contain a private nuget server that requires authentication and it seems the Xenko launcher goes through my entire nuget server configurations and because authentication is required for my private one it fails. In VS2017 i temporarily disabled my server and voila my launcher works, much thanks man!!. @xen2 probably this can be fixed in the launcher to only go through its necessary nuget servers and leave our setups alone :-).. But all things said, to anyone who experience this you know what to do.

**Comment by xen2** [comment]
I could also already reproduce by adding a non working URL in the nuget server list.
Currently taking a look (however the strange thing is that NuGet is doing the repository loop and there is no try-catch, maybe I am supposed to use another API, have to look further).

**Comment by TomGroner** [comment]
I run into similar issues with all VS projects when restoring packages. It will fail even if I am not using any of my own personal / work packages in the project and spin up the server or disable the package source. It's possible there are no hooks to tie into :( This might be a case where the error message mentions what we found here, or has a link to a page with those details, etc.

**Comment by xen2** [comment]
Fixed. Note that even running `nuget.exe install Newtonsoft.Json` fails if there is a non-valid nuget source (but Visual Studio is OK with it)

**Comment by Simba-Mupfunya** [comment]
Thanks @xen2

**Comment by Simba-Mupfunya** [comment]
Just so you know  after turing of my  Telerik nuget package source which required authentication ofcource, i updated to the new version BUT as you will see for the Launcher error message below you are still trying toprocess these nuget package sources that arent Xenko

`An error occurred while updating the launcher. If the problem persists, please reinstall this application.
Details: NuGet.Protocol.Core.Types.FatalProtocolException: An error occurred while retrieving package metadata for 'Xenko.Launcher' from source 'Telerik NuGet'. ---> NuGet.Protocol.Core.Types.FatalProtocolException: The V2 feed at 'https://nuget.telerik.com/nuget/FindPackagesById()?id='Xenko.Launcher'&semVerLevel=2.0.0' returned an unexpected status code '401 Please provide Authorization headers with your request.'. at NuGet.Protocol.V2FeedParser.<>cDisplayClass50_0.<<LoadXmlAsync>b1>d.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.Protocol.HttpSource.<ProcessResponseAsync>d14`1.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.Protocol.V2FeedParser.<LoadXmlAsync>d50.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.Protocol.V2FeedParser.<QueryV2FeedAsync>d49.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.Protocol.V2FeedParser.<FindPackagesByIdAsync>d38.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.Protocol.DependencyInfoResourceV2Feed.<ResolvePackages>d5.MoveNext() --- End of inner exception stack trace --- at NuGet.Protocol.DependencyInfoResourceV2Feed.<ResolvePackages>d5.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.PackageManagement.NuGetPackageManager.<GetLatestVersionCoreAsync>d91.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.PackageManagement.NuGetPackageManager.<>cDisplayClass90_1.<<GetLatestVersionAsync>b0>d.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.PackageManagement.NuGetPackageManager.<GetLatestVersionAsync>d90.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Xenko.Core.Packages.NugetStore.<GetUpdates>d85.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Xenko.LauncherApp.Services.SelfUpdater.<UpdateLauncherFiles>d4.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Xenko.LauncherApp.Services.SelfUpdater.<>cDisplayClass3_0.<<SelfUpdate>b0>d.MoveNext()`

**Comment by Kryptos-FR** [comment]
What do you mean by "disabling my Telerik nuget package source"?

**Comment by Simba-Mupfunya** [comment]
In vs2017 i just have to uncheck it from my list of package sources and by doing that it allows the xenko launcher to work again
